json/codec.go

```diff
@@ -169,9 +172,6 @@ func constructCodec(t reflect.Type, seen map[reflect.Type]*structType) (c codec)
 
 	case reflect.Ptr:
 		c = constructPointerCodec(t, seen)
-
-	default:
-		c = constructUnmarshalTypeErrorCodec(t)
```

Comment:
Don't we want to keep the catch-all error case?

Reply:
I replaced it below with checks for `c.encode == nil` and `c.decode == nil`, but yes this one is probably more clear, I'll revert 👍